File: Web/src/Entity/Categorie.php
<?php

namespace App\Entity;

use App\Repository\CategorieRepository;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * @ORM\Entity(repositoryClass=CategorieRepository::class)
 * @UniqueEntity("nom")
 */
class Categorie
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer", nullable=true)
     * @Groups("categorie")
     */
    private $id;

    /**
     * @Assert\NotBlank(message="Veuillez remplir tous les champs")
     * @ORM\Column(type="string", length=255, nullable=true)
     * @Groups("categorie")
     */
    private $nom;

    /**
     * @Assert\NotBlank(message="Veuillez remplir tous les champs")
     * @ORM\Column(type="string", length=255, nullable=true)
     * @Groups("categorie")
     */
    private $boitevitesse;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getBoiteVitesse(): ?string
    {
        return $this->boitevitesse;
    }

    public function setBoiteVitesse(?string $boitevitesse): self
    {
        $this->boitevitesse = $boitevitesse;

        return $this;
    }
}

File: Mobile/src/com/esprit/app/entity/Categorie.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.esprit.app.entity;

public class Categorie {
    private int id;
    private String nom;
    private String boitevitesse;

    public Categorie(int id, String nom, String boitevitesse) {
        this.id = id;
        this.nom = nom;
        this.boitevitesse = boitevitesse;
    }

    public Categorie(String nom, String boitevitesse) {
        this.nom = nom;
        this.boitevitesse = boitevitesse;
    }

    public Categorie() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getBoitevitesse() {
        return boitevitesse;
    }

    public void setBoitevitesse(String boitevitesse) {
        this.boitevitesse = boitevitesse;
    }

    @Override
    public String toString() {
        return "Categorie{" + "id=" + id + ", nom=" + nom + ", boitevitesse=" + boitevitesse + '}';
    }
    
    
}
